src/lib/import-export/import/import-manager.ts

```diff
@@ -31,35 +31,90 @@ export function selectImporter(
+	backupFile: BackupArchiveInfo,
+	extractionDirectory: string,
+	onEvent: ( data: ImportExportEventData ) => void,
+	options: ImporterOption[]
+) {
+	const backupHandler = BackupHandlerFactory.create( backupFile );
+	const fileList = await backupHandler.listFiles( backupFile );
+	const importer = selectImporter( fileList, extractionDirectory, onEvent, options );
+	if ( importer ) {
+		await backupHandler.extractFiles( backupFile, extractionDirectory );
+	}
+
+	return { importer, backupHandler };
+}
+
+async function createExtractionDirectory() {
+	const extractionDirectory = await fsPromises.mkdtemp( path.join( os.tmpdir(), 'studio_backup' ) );
+	const removeExtractionDirectory = async () => {
+		await fsPromises.rm( extractionDirectory, { recursive: true } );
+	};
+	return { extractionDirectory, removeExtractionDirectory };
+}
+
 export async function importBackup(
 	backupFile: BackupArchiveInfo,
 	sitePath: string,
 	onEvent: ( data: ImportExportEventData ) => void,
 	options: ImporterOption[]
 ): Promise< ImporterResult > {
-	const extractionDirectory = await fsPromises.mkdtemp( path.join( os.tmpdir(), 'studio_backup' ) );
-	let removeBackupListeners;
-	let removeImportListeners;
+	const { extractionDirectory, removeExtractionDirectory } = await createExtractionDirectory();
+	let removeBackupListeners: ( () => void ) | undefined;
+	let removeImportListeners: ( () => void ) | undefined;
+
 	try {
-		const backupHandler = BackupHandlerFactory.create( backupFile );
-		const fileList = await backupHandler.listFiles( backupFile );
-		const importer = selectImporter( fileList, extractionDirectory, onEvent, options );
+		const { importer, backupHandler } = await extractBackupFile(
+			backupFile,
+			extractionDirectory,
+			onEvent,
+			options
+		);
 
-		if ( importer ) {
-			removeBackupListeners = handleEvents( backupHandler, onEvent, BackupExtractEvents );
-			removeImportListeners = handleEvents( importer, onEvent, ImporterEvents );
-			await backupHandler.extractFiles( backupFile, extractionDirectory );
-			return await importer.import( sitePath );
-		} else {
+		if ( ! importer ) {
 			throw new Error( 'No suitable importer found for the given backup file' );
 		}
+
+		removeBackupListeners = handleEvents( backupHandler, onEvent, BackupExtractEvents );
+		removeImportListeners = handleEvents( importer, onEvent, ImporterEvents );
+
+		return await importer.import( sitePath );
 	} catch ( error ) {
 		console.error( 'Backup import failed:', ( error as Error ).message );
 		throw error;
 	} finally {
 		removeBackupListeners?.();
 		removeImportListeners?.();
-		await fsPromises.rm( extractionDirectory, { recursive: true } );
+		await removeExtractionDirectory();
+	}
+}
+
+export async function getMetaFromBackupFile( backupFile: BackupArchiveInfo ) {
+	const { extractionDirectory, removeExtractionDirectory } = await createExtractionDirectory();
+	// eslint-disable-next-line @typescript-eslint/no-empty-function
+	const silentOnEvent = () => {};
```

Comment:
If the backup file is big enough, we could consider reflecting the extraction of the metafile in the import progress bar as it will take some time. Hence, we could consider adding events to track the start and complete steps of the process.